components/server/src/express/ws-handler.ts

```diff
@@ -65,6 +68,10 @@ export class WsExpressHandler {
     ): void {
         const stack = WsLayer.createStack(...handlers);
         const dispatch = (ws: websocket, request: express.Request) => {
+            const toTerminate = Disposable.create(() => ws.terminate());
+            const cancelTerminate = this.disposables.push(toTerminate);
+            ws.on("close", () => cancelTerminate.dispose());
```

Comment:
Dispatch is called when a web socket is opened. We install a disposable to terminate a ws if server is shutting down. But if ws is terminated earlier we need to clean up the disposable so we listen on `close`.

Maybe I misunderstand then `dispatch` is called?

Reply:
@akosyakov I checked: As we have a liveness- but no readiness probe configured, this will work on SIGTERM, because on pod delete it will mark the pod as "not ready" anymore.

The more general approach would be to have a process wide "shutdown" mechanism/state which is set when before calling dispose, which influences the response of a custom "readiness" probe.

But that is out of scope here. :+1: